includes/class-wcpdf-admin.php

```diff
@@ -725,11 +770,25 @@ public function regenerate_document() {
 		$order_id = absint( $_POST['order_id'] );
 		$order = WCX::get_order( $order_id );
 		$document_type = sanitize_text_field( $_POST['document_type'] );
-		$form_data = json_decode( stripslashes( $_POST['form_data'] ), true );
+
+		// parse form data
+		parse_str($_POST['form_data'], $form_data);
+		if ( is_array( $form_data ) ) {
+			foreach ( $form_data as $key => &$value ) {
+				if ( is_array( $value ) && !empty( $value[$order_id] ) ) {
+					$value = $value[$order_id];
+				}
+			}
+		}
 
 		try {
 			$document = wcpdf_get_document( $document_type, wc_get_order( $order_id ) );
 			if ( !empty($document) && $document->exists() ) {
+				// perform legacy date fields replacements check
+				if( in_array( $document->get_type(), array( 'proforma', 'credit-note' ) ) && ! is_array( $form_data["_wcpdf_{$document->slug}_date"] ) ) {
```

Comment:
this shouldn't be limited to proforma and credit-note, could be third party documents as well. It should also have a check for whether `_wcpdf_{$document->slug}_date` isset at all (form data could be empty just regenerating).